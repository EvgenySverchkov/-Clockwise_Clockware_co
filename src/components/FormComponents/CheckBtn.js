import React from "react";
import PropTypes from "prop-types";

const CheckBtn = ({ id, name, changeHandler, nm }) => (
<div key={id + 1} className="form-check-inline">
    <label className="form-check-label" htmlFor={name}>
      <input
        type="checkbox"
        className="form-check-input towns"
        id={name}
        value={name}
        name = {nm}
        onChange = {changeHandler}
      />
      {name}
    </label>
  </div>
);

CheckBtn.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};

export default CheckBtn;
