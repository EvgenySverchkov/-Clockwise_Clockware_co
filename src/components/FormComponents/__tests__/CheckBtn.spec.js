import React from 'react';
import {create} from "react-test-renderer";

import CheckBtn from "../CheckBtn";

it("<CheckBtn /> test which is used in forms", ()=>{
    const mockProps = {
        id: 1, 
        name: "Name", 
        changeHandler: jest.fn(), 
        nm: "Nm"
    }
    const component = create(<CheckBtn {...mockProps}/>)
    expect(component.toJSON()).toMatchSnapshot();
})