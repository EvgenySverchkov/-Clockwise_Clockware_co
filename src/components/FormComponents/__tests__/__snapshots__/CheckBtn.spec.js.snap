// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBtn /> test which is used in forms 1`] = `
<div
  className="form-check-inline"
>
  <label
    className="form-check-label"
    htmlFor="Name"
  >
    <input
      className="form-check-input towns"
      id="Name"
      name="Nm"
      onChange={[MockFunction]}
      type="checkbox"
      value="Name"
    />
    Name
  </label>
</div>
`;
